### PR TITLE
Add NoOpBootstrapIndex config for hudi table

### PR DESCRIPTION
**Pull request by simon824**
Add create table options for DataLakeSuiteMixin

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @pan3793 @yanghua

**Comment by pan3793** [comment]
Have a second thought on it, I change my mind now, because it seems that `tableOptions` is not a common field for DDL, as we create several tables in different methods using different `options`.

But as we disscue before, it's valuable to set `hoodie.bootstrap.index.class=org.apache.hudi.common.bootstrap.index.NoOpBootstrapIndex` for hudi tables.

@simon824 Sorry for making you back and forth.

**Comment by simon824** [comment]
> Have a second thought on it, I change my mind now, because it seems that `tableOptions` is not a common field for DDL, as we create several tables in different methods using different `options`.
> 
> But as we disscue before, it's valuable to set `hoodie.bootstrap.index.class=org.apache.hudi.common.bootstrap.index.NoOpBootstrapIndex` for hudi tables.
> 
> @simon824 Sorry for making you back and forth.

what about separate common config like `hoodie.bootstrap.index.class`  and custom config like `primaryKey`?

https://github.com/apache/incubator-kyuubi/pull/1040/files#diff-98f5ef031c8c38b4be598220eaa8c0f8c23522b32fa8c63fdcf71e46f44a9dacR42-R52

**Review by pan3793 — APPROVED** [review]

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/BasicHudiJDBCTests.scala [hunk 1]
```diff
@@ -75,13 +75,14 @@ trait BasicHudiJDBCTests extends JDBCTestUtils with HudiSuiteMixin {
            | ) using $format
            | options (
            |   primaryKey = 'id',
-           |   preCombineField = 'ts'
+           |   preCombineField = 'ts',
+           |   hoodie.bootstrap.index.class =
```
Comment (pan3793) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/BasicHudiJDBCTests.scala:79]: remove the space at end of line

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**